src/main/java/io/strimzi/kafka/bridge/http/HttpOpenApiOperation.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.bridge.http;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Represents an OpenApi operation with related logging
+ */
+public abstract class HttpOpenApiOperation implements Handler<RoutingContext> {
+
+    protected static final Logger log = LoggerFactory.getLogger(HttpOpenApiOperation.class);
+
+    protected final HttpOpenApiOperations openApiOperation;
+    protected final Level logLevel;
+
+    public HttpOpenApiOperation(HttpOpenApiOperations openApiOperation, Level logLevel) {
+        this.openApiOperation = openApiOperation;
+        this.logLevel = logLevel;
+    }
+
+    public abstract void process(RoutingContext routingContext);
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        this.logRequest(routingContext);
```

Comment:
Why not make the call to `httpBridgeContext.setOpenApiOperation()` here too, to avoid each handler method from having to do that?

Reply:
I was thinking about that but we need to pass the `httpBridgeContext` instance to the operation because it's a separated class while the `httpBridgeContext` instance is in the `HttpBridge`.

Reply:
I guess it worked in my prototype because I was lazy and used a member class.

Reply:
You weren't lazy but yes this is the reason why it worked :-)